sunpycube/cube/datacube.py

```diff
@@ -193,7 +193,7 @@ def _choose_wavelength_slice(self, offset):
         if self.data.ndim == 4:
             raise cu.CubeError(4, "Can only work with 3D cubes")
 
-        axis = 1 if self.axes_wcs.wcs.ctype[-1] in ['TIME', 'UTC'] else 0
+        axis = -2 if self.axes_wcs.wcs.ctype[0] in ['TIME', 'UTC'] else -1
```

Comment:
ctype[0] will be time so changing from 1 to -2 as orientation is changed. similarly for -1.